lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/ssec/AbstractTestS3FileSystemWithSseC.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.s3.ssec;
+
+import io.trino.filesystem.s3.AbstractTestS3FileSystem;
+import software.amazon.awssdk.utils.BinaryUtils;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public abstract class AbstractTestS3FileSystemWithSseC
```

Comment:
I don't think this abstraction is needed just to share `generateCustomerKey()`. Also it looks like this class is used only once.